json_serializable/lib/src/default_container.dart

```diff
@@ -27,6 +29,10 @@ class DefaultContainer {
       return input._result(defaultValue);
     }
 
+    if (input is LambdaResult && defaultValue != null) {
```

Comment:
`LambdaResult` is used by the converters – so we can catch it here and handle the default case!